hooks/useLoginHook.ts

```diff
@@ -0,0 +1,158 @@
+          field: 'isUltraUser',
+          value: userProfile.plan === 'ultra',
+        });
+        dispatch({
+          field: 'hasMqttConnection',
+          value: userProfile.hasMqttConnection,
+        });
+        dispatch({
+          field: 'isConnectedWithLine',
+          value: userProfile.isConnectedWithLine,
+        });
+        dispatch({
+          field: 'user',
+          value: {
+            id: session.user.id,
+            email: session.user.email,
+            plan: userProfile.plan || 'free',
+            token: session.access_token,
+            referralCode: userProfile.referralCode,
+            referralCodeExpirationDate: userProfile.referralCodeExpirationDate,
+            proPlanExpirationDate: userProfile.proPlanExpirationDate,
+            hasReferrer: userProfile.hasReferrer,
+            hasReferee: userProfile.hasReferee,
+            isInReferralTrial: userProfile.isInReferralTrial,
+          },
+        });
+
+        updateUserInfo({
+          id: userProfile.id,
+          email: userProfile.email,
+          plan: userProfile.plan || 'free',
+          associatedTeacherId: userProfile.associatedTeacherId,
+          isTeacherAccount: userProfile.isTeacherAccount,
+          isTempUser: userProfile.isTempUser,
+          tempUserUniqueId: userProfile.tempUserUniqueId,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
+        toast.error(
+          t('Unable to load your information, please try again later.'),
+        );
+      })
+      .finally(() => {
+        dispatch({ field: 'appInitialized', value: true });
+      });
+  };
+
+  const checkSurveyFilled = (session: Session) => {
+    supabase
+      .from('user_survey')
+      .select('name')
+      .eq('uid', session.user.id)
+      .then(({ data }) => {
+        if (!data || data.length === 0) {
+          dispatch({ field: 'isSurveyFilled', value: false });
+        } else {
+          dispatch({ field: 'isSurveyFilled', value: true });
+        }
+      });
+  };
+
+  useEffect(() => {
+    const { data } = supabase.auth.onAuthStateChange((event, session) => {
+      if (event === 'INITIAL_SESSION') {
+        // handle initial session
+        if (!session) {
+          dispatch({ field: 'appInitialized', value: true });
+        } else {
+          handleUserProfileUpdate(session);
+          checkSurveyFilled(session);
+        }
+      } else if (event === 'SIGNED_IN') {
+        if (session?.user && !user) {
+          // User info has been updated for this session
+          handleUserProfileUpdate(session);
+          checkSurveyFilled(session);
+        }
+      } else if (event === 'SIGNED_OUT') {
+        // handle sign out event
+      } else if (event === 'PASSWORD_RECOVERY') {
+        // handle password recovery event
+      } else if (event === 'TOKEN_REFRESHED') {
+        // handle token refreshed event
+      } else if (event === 'USER_UPDATED') {
+        // handle user updated event
+      }
```

Comment:
I wonder do we need to handle these events? Or you already have the expired token handle somewhere else?

Reply:
Let me remove them. The expired token is handled automatically.